essentials-framework/Essentials Core/PepperDashEssentialsBase/Routing/RoutingInterfaces.cs

```diff
@@ -111,10 +111,70 @@ public interface IRmcRouting : IRoutingNumeric
         IntFeedback AudioVideoSourceNumericFeedback { get; }
     }
 
-	/// <summary>
+
+    /// <summary>
+    /// Defines an IRmcRouting with a feedback event 
+    /// </summary>
+    public interface ITxRoutingWithFeedback : ITxRouting, IRoutingNumericFeedback
+    {
+    }
+
+    /// <summary>
+    /// Defines an IRmcRouting with a feedback event 
+    /// </summary>
+    public interface IRmcRoutingWithFeedback : IRmcRouting, IRoutingNumericFeedback
+    {
+    }
+
+    /// <summary>
 	/// Defines an IRoutingOutputs devices as being a source - the start of the chain
 	/// </summary>
 	public interface IRoutingSource : IRoutingOutputs
 	{
 	}
+
+    /// <summary>
+    /// Defines an event structure for reporting output route data
+    /// </summary>
+    public interface IRoutingNumericFeedback : IKeyName
+    {
+        event EventHandler NumericSwitchChange;
+        void OnSwitchChange(RoutingNumericEventArgs e);

```

Comment:
We don't need to add the OnSwitchChange method to the interface. Only public methods should be added, and the event fired in whatever manner the implementer of the interface wants. Change the EventHandler to be something like `event EventHandler<RoutingNumericEventArgs> NumericSwitchChange` and it'll work just fine. Having the `OnSwitchChange` method as a public method on the class could lead to spurious events if someone calls the method randomly.